google/resource_storage_bucket.go

```diff
@@ -77,6 +79,65 @@ func resourceStorageBucket() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"action": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Set:      resourceGCSBucketLifecycleRuleActionHash,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"storage_class": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"condition": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Set:      resourceGCSBucketLifecycleRuleConditionHash,
```

Comment:
Same as above, the XML API doc looks a bit more precise than the JSON API about those facts.

XML API:

> Condition	Defines the condition under which the action will be taken, which must contain at least one specific condition element. Required.

JSON API:

> The condition(s) under which the action will be taken.